core/src/main/java/org/mskcc/cbio/portal/servlet/PatientView.java

```diff
@@ -484,12 +485,39 @@ private void setClinicalInfo(HttpServletRequest request) throws DaoException {
         }
         
         // path report
-        String typeOfCancer = cancerStudy.getTypeOfCancerId();
-        if (patientId!=null && patientId.startsWith("TCGA-")) {
-            String pathReport = getTCGAPathReport(typeOfCancer, patientId);
-            if (pathReport!=null) {
-                request.setAttribute(PATH_REPORT_URL, pathReport);
+        if (GlobalProperties.useInternalPathReports()) {
+            // Expected url path 'studyId/patientId/sampleId.pdf' maps to expected filesystem path 'studyId.patientId.sampleId.pdf'
+            final String reportBaseURL = "pathology_report/" + cancerStudy.getCancerStudyStableId() 
+                + "/" + patientId + "/";
+            final String reportBasePath = GlobalProperties.getInternalPathReportRoot() + "/" 
+                + cancerStudy.getCancerStudyStableId() + "." + patientId + ".";
+            if (isPatientView) {
+                Map<String,String> sampleReportUrls = new HashMap<String,String>();
+                for (String sample : samples) {
+                    final String reportFileName = sample + ".pdf";
+                    if (new File(reportBasePath + reportFileName).exists()) {
+                        sampleReportUrls.put(sample, reportBaseURL + reportFileName);
+                    }
+                }
+                if (!sampleReportUrls.isEmpty()) {
+                    //ToDo: send across the map of all sample URLs and handle the map within the patient_view.jsp and the path_report.jsp
+                    final String firstSampleReport = sampleReportUrls.entrySet().iterator().next().getValue();
```

Comment:
Currently the patient view only shows the pathology report for its first sample that contains a pathology report. However the intention is to (at some point) display all sample pathology reports within the patient view. The mapping code creates a map of sample ids to their pathology report url for each of the samples within the study that have a pathology report on the filesystem. Since only the first sample report is shown within the patient view the mapping could be modified to only check study samples until one is found with an existing report, however since it is intended to show all of the sample reports within the patient view (but we just don't know how they want it displayed at the moment) it seems reasonable to keep this code in that finds all of the sample pathology reports.